event/brokerPool.go

```diff
@@ -0,0 +1,130 @@
+package event
+
+import (
+	"context"
+	"sync"
+
+	"github.com/sirupsen/logrus"
```

Comment:
Should it depend on `logrus`?
Not sure what is the best way to surface background errors.

Reply:
I'd say it's not ideal either, so we in the future we could either abstract this by having a project-wide "error handler" or even "logger" abstraction, but for now it is also not a regression since a couple of our other facilitations do still depend on `logrus` as well 😬